extending-valtimo/plugin/custom-plugin-definition.md

```diff
@@ -82,6 +82,57 @@ class TwitterPluginFactory(
 }
 ```
 
+### Plugin categories
+There are use-cases where you would like to use another plugin in order to share functionality and/or configuration.
+Plugin Categories to the rescue: this functionality lets you choose a plugin implementing a specific interface into your plugin as a `@PluginProperty` in the configuration screen.
+
+For example, you might want to use a supplier for your tweets:
+```kotlin
+@PluginCategory()
+interface TweetSupplier {
+    fun supplyMessage() : String
+}
+```
+
+```kotlin
+@Plugin(
+  key = "property-tweet-supplier",
+  title = "Property tweet supplier",
+  description = "Get a message from a property"
+)
+class PropertyTweetSupplier: TweetSupplier {
+    @PluginProperty(key = "message", secret = false)
+    private lateinit var message: String
+    
+    fun supplyMessage(): String {
+       return message
+    }
+}
+```
+
+```kotlin
```

Comment:
description example:

When creating a configuration of the TwitterPlugin the front-end should get and show a list of all available configuration of type x. The id of the chosen configuration will be part of the properties submitted for the creation of the TwitterPlugin configuration. The PluginProperty can reference the interface type corresponding to the category. The plugin will then be automatically injected when using the TwitterPlugin.